jetstream_pt/engine.py

```diff
@@ -545,10 +556,13 @@ def generate(
   ) -> tuple[DecodeState, engine_api.ResultTokens]:
     # seq_len = padded_tokens.shape[0]
     pos = decode_state.current_position
-    input_indexes = jnp.full((1,), pos)
-
-    # fill mask first
-    mask = decode_state.mask.at[:, decode_state.current_position].set(0)
+    if self.env.ring_buffer:
+      input_indexes = jnp.full((1,), pos)
```

Comment:
What if we change current_position to [batch_size, 1], can we use same logic do mask for both ring_buffer and onn_ring_buffer?

Reply:
Yes. I mean if we change current_position to [batch_size, 1], different slot can have different the current_position.  For non ring buffer case, the current_position should be same as input_pos.

Reply:
It will cause performance regression. Please check jax_experiments.py/test7, inserting with batching + position array takes much longer, like x4~x5